corehq/apps/settings/forms.py

```diff
@@ -263,21 +263,31 @@ def __init__(self, **kwargs):
 
 
 class HQApiKeyForm(forms.Form):
+    ALL_DOMAINS = '*'
     name = forms.CharField()
     ip_allowlist = SimpleArrayField(
         forms.GenericIPAddressField(),
         label=ugettext_lazy("Allowed IP Addresses (comma separated)"),
         required=False
     )
+    domain = forms.ChoiceField(
+        required=False,
+        help_text=ugettext_lazy("Limit the key's access to a single project space")
+    )
 
     def __init__(self, *args, **kwargs):
+        self.couch_user = kwargs.pop('couch_user')
         super().__init__(*args, **kwargs)
 
+        user_domains = self.couch_user.get_domains()
+        all_domains = (self.ALL_DOMAINS, _('All Domains'))
```

Comment:
Minor: user-facing content usually says "project" or "project space" instead of "domain."

Reply:
good catch. https://github.com/dimagi/commcare-hq/pull/28493/commits/be674a9d6e3a4750ece38898b580b97dfaf506d3